_raml/apps/products/products.raml

```diff
@@ -130,23 +130,51 @@ post:
     responses:
       204:
 
-  /crossselling:
-    displayName: Product crossselling
-    get:
-      is: [ locale, currency ]
-      queryParameters:
-        includeInvisible:
-          description: Includes invisible products. Only available with `products_write` authorisation.
-          type: boolean
-          required: false
-      description: Returns information on the crossselling of products. *epagesNow
-      securedBy: [products_read]
-      responses:
-        200:
-          body:
-            application/json:
-              example: !include examples/get_product_productid_crossselling_response.json
-              schema: !include schema/get_product_productid_crossselling.schema.json
+/crossselling:
+  displayName: Product crossselling
+  get:
+    is: [ locale, currency ]
+    queryParameters:
+      includeInvisible:
+        description: Includes invisible products. Only available with `products_write` authorisation.
+        type: boolean
+        required: false
+    description: Returns information on the crossselling of products. *epagesNow
+    securedBy: [products_read]
+    responses:
+      200:
+        body:
+          application/json:
+            example: !include examples/get_product_productid_crossselling_response.json
+            schema: !include schema/get_product_productid_crossselling.schema.json
+
+  put:
+    description: Replace all cross selling assignments
+    securedBy: [products_write]
+    body:
+      application/json:
+        example: !include examples/put_product_productid_crossselling_request.json
+        schema: !include schema/put_product_productid_crossselling_request_schema.json
+    responses:
+      200:
+        body:
+          application/json:
+            example: !include examples/get_product_productid_crossselling_response.json
+            schema: !include schema/get_product_productid_crossselling.schema.json
+
+  patch:
+    description: Add/Remove cross selling assignments
```

Comment:
To match the wording of other `PATCH` requests I would suggest to change it as follows:
```suggestion
    description: Updates information on the cross-selling assignments for a single product with partial data according to [RFC 6902](https://tools.ietf.org/html/rfc6902). You can either add or remove certain attributes. *epagesNow
```
Furthermore, we need to add the information to which product this feature applies. Is it just *epagesNow, as the `GET` request, or also base?
However, I'm really unsure about the part "with partial data according to [RFC 6902](https://tools.ietf.org/html/rfc6902)." Is this applicable for this endpoint? I'm also wondering, if we need to add information about supported JSON-PATCH operations and attribute paths that allow updates via PATCH (see [example](https://developer.epages.com/apps/api-reference/patch-shopid-orders-orderid.html))

Reply:
I'm fine with your wording, and the link to 'RFC 6902' doesn't hurt, although I have made a minor adjustment in 'add or remove certain attributes' to 'add or remove assignments'.
Because we're dealing with assignments, there is only one path and we already state that only add/remove are supported. Coupled with this API being a fairly niche one along with the example request, and I think it is ok with your update.